File: android/app/src/main/java/com/r2r/readytorevise/data/remote/dto/SelfGradeReviewDto.kt
package com.r2r.readytorevise.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class SelfGradeReviewDto(
    val studyItemId: String,
    val responseTimeMs: Int? = null,
    val sessionId: String? = null
)

File: backend/src/modules/reviews/presentation/dto/self-grade-review.dto.ts
import { IsInt, IsOptional, IsUUID, Min } from 'class-validator';

export class SelfGradeReviewDto {
    @IsUUID()
    studyItemId!: string;

    @IsOptional()
    @IsInt()
    @Min(0)
    responseTimeMs?: number;

    @IsOptional()
    @IsUUID()
    sessionId?: string;
}
